src/SME.SGP.Aplicacao/Comandos/ComandosAtividadeAvaliativa.cs

```diff
@@ -147,10 +148,13 @@ public async Task Excluir(long idAtividadeAvaliativa)
 
             var disciplinaId = long.Parse(atividadeDisciplinas.FirstOrDefault().DisciplinaId);
 
-            var regenteAtual = await mediator.Send(new ObterUsuarioPossuiPermissaoNaTurmaEDisciplinaQuery(long.Parse(atividadeDisciplinas.FirstOrDefault().DisciplinaId), turma.CodigoTurma, DateTime.Now.Date, usuario));
-
-            var aula = await repositorioAula.ObterAulas(turma.CodigoTurma, atividadeAvaliativa.UeId, regenteAtual ? string.Empty : usuario.CodigoRf, atividadeAvaliativa.DataAvaliacao, atividadeDisciplinas.Select(s=> s.DisciplinaId).ToArray(), usuario.EhProfessorCj());
+            var usuarioPossuiAtribuicaoNaTurmaNaData = await mediator.Send(new ObterUsuarioPossuiPermissaoNaTurmaEDisciplinaQuery(disciplinaId, turma.CodigoTurma, atividadeAvaliativa.DataAvaliacao.Date, usuario));
+            
+            var aula = await repositorioAula.ObterAulas(turma.CodigoTurma, atividadeAvaliativa.UeId, usuarioPossuiAtribuicaoNaTurmaNaData ? string.Empty : usuario.CodigoRf, atividadeAvaliativa.DataAvaliacao, atividadeDisciplinas.Select(s=> s.DisciplinaId).ToArray(), usuario.EhProfessorCj());
```

Comment:
Validar atribuição na turma apenas quando for professor;
Manter a verificação com a data atual.

Reply:
Ajustado